packages/toolpad-app/src/appDom/index.ts

```diff
@@ -402,20 +403,18 @@ export function getChildNodes<N extends AppDomNode>(dom: AppDom, parent: N): Nod
 }
 
 export function getParent<N extends AppDomNode>(dom: AppDom, child: N): ParentOf<N> {
+  // Make sure we're using the last version of child in the dom
+  child = getNode(dom, child.id, child.type) as N;
```

Comment:
Should this ideally not be necessary?
Nothing against this extra check but if we plan on the DOM always being up to date maybe we should have a TODO comment to get rid of it at some point.
Or maybe the function arguments should be `childId` and `childType` but that might be too much of a refactor...

Reply:
> Should this ideally not be necessary?

Ideally, yes, but this is caused by the drag & drop code, adding a page row and then the item to the row. In the second call, the row doesn't have a parent yet even though it's added to the dom, the node that was passed to this function is the created one, without a parent.
The easy solution seemed to be to just extract the latest version of the node from the dom. I want to avoid messing to much with the drag&drop code until the paper component updates are in.
Ideally these methods take a `NodeId` instead of an `AppNode` because the latter may be outdated.

Reply:
Sounds good, yeah I was wondering if that kind of logic in the drag & drop is what this was dealing with.
Ok so with the upcoming changes this kind of thing should improve.

Reply:
> Ok so with the upcoming changes this kind of thing should improve.

Yes, additionally I'd like to switch this back to operating on `NodeId`s to make this problem completely go away.